### ISSUE TITLE
Using header= in CCDData intializer raises an error...

### ISSUE THREAD

**Issue by mwcraig** [astropy/ccdproc#145]
...because the argument is passed on to ``NDData``, which doesn't recognize the argument.

The fix would do something like:

+ pop ``header`` from the input dictionary before calling ``super....__init__`` in the ``CCDData`` initializer
+ set self.header to header ini ``CCDData.__init__``

**Comment by mwcraig** [comment]
To be a little more specific, this call raised an error:

```
ccd = CCDData(some_data, header={}, unit='adu')
```
even though the nearly identical call below works:
```
ccd = CCDData(some_data, meta={}, unit='adu')
```

**Comment by mwcraig** [comment]
Closed by #200
